startup/csx1/plans/dark_ff.py

```diff
@@ -6,6 +6,35 @@
 from ..startup.detectors import fccd
 
 
+def ct_flatfield(num_imgs, detectors):
+    """Collect flatfield images for fccd with custom 'plan_name'='count_flatfield'.
+
+The pre-count number of images preserved.
+
+    Parameters
+    -----------
+    num_imgs: int
+
+        Number of images to be measured. If different from current
+        setting, the number of images will revert back to the original
+        after the scan is complete.
+
+    detectors: list
+        List of detectors to be recorded.
+        Default = [fccd]
+
+    """
+    if detectors is None:
+        detectors = [fccd]
+    revert = False
+    num_imgs_initial = yield from bps.rd(fccd.cam.num_images)
+    if num_imgs != num_imgs_initial:
+        revert = True
+        yield from mv(fccd.cam.num_images, num_imgs)
+    yield from count(detectors, md={'plan_name':'count_flatfield'}) #TODO add exception handling
```

Comment:
What sort of exceptions are you anticipating or finding regularly that you want to handle? (This is more a question for my own learning than anything else).

Reply:
User makes a mistake:

-  incorrect number of images, exposure time, 
-  valve is closed or beam is blocked 
-  they want to take darks first because they drastically changed the exposure time  (which means the statistics for the camera are not quite right so they want to make sure the camera settings are appropriate before wasting a  bunch of time collect data that is not ideal)

Reply:
Ahh I see, so something that looks at the User action and raises Exceptions, as opposed to wrapping `count` in
```python
try:
    count(...)
except ... :
    ....
```

Reply:
yep.  likely creating to "sub_plans" like
```
plan = count(......)
reset = mv(fccd.c....)

try 
    yield from plan
except...
```

I have examples of things I've done already that seem to work without having to do fancy stuff. I just didn't have the time to look into the details. I had a look at exception documentation, but it quickly gets theoretical with customize REs.  If the plan was more complicated, I know I cannot expect the pseudo code above to work well.